src/app/shared/services/jokes.service.ts

```diff
@@ -89,10 +95,51 @@ export class JokesService {
     }
   }
 
-  public checkIfJokeIsFavorite(joke: JokeInterface): JokeInterface {
-    this.favoriteJokesService.allJokes.map((favoriteJoke: JokeInterface) => {
-      return favoriteJoke.id === joke.id ? joke = { ...favoriteJoke } : joke;
+  public checkIfJokeIsFavorite(joke: JokeModel): JokeModel {
+    this.favoriteJokesService.allJokes.map((favoriteJoke: JokeModel) => {
+      return favoriteJoke.id === joke.id ? (joke = { ...favoriteJoke }) : joke;
     });
     return joke;
   }
+
+  public getJokeById(id: number): Observable<JokeModel> {
+    return this.http.get<JokeModel>(`${this.apiUrl}/${id}`);
+  }
+
+  public createJoke(joke: JokeModel): Observable<JokeModel> {
+    return this.http.post<JokeModel>(this.apiUrl, joke);
+  }
+
+  public updateJoke(joke: JokeModel): Observable<JokeModel[]> {
+    return this.http.put<JokeApiModel>(this.apiUrl, joke).pipe(
+      map((jokes: JokeApiModel) => {
+        return this.jokesMapperService.mapJokeApiForJokes([jokes]);
+      })
+    );
+  }
+
+  public deleteJoke(id: number | string): Observable<any> {
```

Comment:
What is the difference between `deleteJoke` and `deleteJokeById` methods?

Reply:
deleteJoke is clearly API call method
deleteJokeById deletes joke from saved jokes in service